### ISSUE TITLE
Aliases containig logical subprocess operators not handled properly

### ISSUE THREAD

**Issue by con-f-use** [xonsh/xonsh#3270]
Consider these three aliases:
```
┌─12-15:40 [confus@confusion /tmp/tmp.w8lR957tJh]                           😱  😱
└$ tail ~/.xonshrc -n8
# Working alias:
aliases['envpy2']='echo "layout python2" > .envrc; direnv allow .'

# Broken alias:
aliases['envpy3']='echo "layout python3" > .envrc && direnv allow .'

# Even more broken alias:
aliases['envnorrr']='echo "layout python2" && echo test'
```
When executing these, only the py2 one behaves as expected:
```
┌─12-15:40 [confus@confusion /tmp/tmp.w8lR957tJh]                           😱  😱
└$ envpy2                                                                                                                                                                        
direnv: loading .envrc
New python executable in /tmp/tmp.w8lR957tJh/.direnv/python-2.7.15+/bin/python2
Also creating executable in /tmp/tmp.w8lR957tJh/.direnv/python-2.7.15+/bin/python
Installing setuptools, pip, wheel...
done.
Running virtualenv with interpreter /usr/bin/python2
direnv: export +VIRTUAL_ENV -PS2 ~PATH

┌─12-15:40 [confus@confusion /tmp/tmp.w8lR957tJh]                           😱  😱
└$ envpy3                                                                                                                                                                        
layout python3 > .envrc anddirenv allow .

┌─12-15:40 [confus@confusion /tmp/tmp.w8lR957tJh]                           😱  😱
└$ envnorrr                                                                                                                                                                      
layout python2 andecho test
```

The others are broken because of  the logical operator between commands. Also stream redirection, which is the difference between `envpy3` and `envnorrr` (note the literal `and` instead of `&&` in alias definition) seems to make them behave differently.

<details>
  <summary>xonsh version and details</summary>

    
    $ xonfig                                                                              
    +------------------+----------------------+
    | xonsh            | 0.9.10               |
    | Git SHA          | 550d70d2             |
    | Commit Date      | May 22 01:42:07 2019 |
    | Python           | 3.6.8                |
    | PLY              | 3.11                 |
    | have readline    | True                 |
    | prompt toolkit   | 2.0.9                |
    | shell type       | prompt_toolkit2      |
    | pygments         | 2.4.0                |
    | on posix         | True                 |
    | on linux         | True                 |
    | distro           | ubuntu               |
    | on darwin        | False                |
    | on windows       | False                |
    | on cygwin        | False                |
    | on msys2         | False                |
    | is superuser     | False                |
    | default encoding | utf-8                |
    | xonsh encoding   | utf-8                |
    | encoding errors  | surrogateescape      |
    +------------------+----------------------+
    
    
    ┌─12-15:40 [confus@confusion /tmp/tmp.w8lR957tJh]                           😱  😱
    └$ alias | grep env
    envpy2=ExecAlias('echo "layout python2" > .envrc; direnv allow .\n', filename='<exec-alias:envpy2>')
    envpy3=ExecAlias('echo "layout python3" > .envrc; direnv allow .\n', filename='<exec-alias:envpy3>')
    envnorrr=['echo', 'layout python2', 'andecho', 'test']
    

</details>

**Comment by scopatz** [comment]
Thanks for reporting! I think this is because this line: `https://github.com/xonsh/xonsh/blob/master/xonsh/aliases.py#L46` does not detect those characters as needing to be an `ExecAlias` automatically.  PRs welcome to expand this! I am on vacation for the next couple days

**Comment by gforsyth** [comment]
Hey @con-f-use -- there's a fix in #3272 if you want to try it out.

**Comment by con-f-use** [comment]
Seems to work. That was quick.
